guides/20240819_history_embedding.md

```diff
@@ -0,0 +1,237 @@
+
+This guide is designed for users who want to gain deeper insights into their browsing habits or need a more powerful way to sift through large amounts of history data. From setting up the necessary tools to executing detailed search queries with various filters, you'll gain the knowledge to fully utilize ChromaDB’s capabilities. Whether you’re a developer looking to integrate this functionality into a larger project or simply someone who wants to better manage their digital footprint, these step-by-step instructions will help you achieve your goals.
+
+### TL;DR
+
+- **Export Browser History**: Use the Export Chrome History extension to get a CSV file.
+- **Set Up Environment**: Install Daytona and create a [workspace](/definitions/d/daytona-workspace).
+- **Embed History**: Run the `search.py` script to embed your history data.
+- **Search History**: Perform searches using various filters and options.
+
+## Preparations
+
+### Installing Prerequisites
+
+Before you begin, ensure you have [Daytona](https://github.com/daytonaio/daytona) installed.
+
+You can install Daytona by running the following command in your terminal:
+
+```bash
+curl -L https://download.daytona.io/daytona/install.sh | sudo bash
+```
+
+Daytona creates a controlled [development environment](/definitions/d/development-environment) where you can work on the project with all dependencies correctly configured, reducing the likelihood of compatibility issues.
+
+### Exporting Browser History
+
+To work with your browser history, you'll need to export it from your browser:
+
+1. **Install the Export Chrome History Extension**: Download and install the [Export Chrome History](https://chrome.google.com/webstore/detail/export-chrome-history/) Chrome extension.
+2. **Export to CSV**: Use the extension to export your browser history as a CSV file. Save it to a convenient location, like `~/Downloads/history.csv`.
+
+Example CSV snippet:
+
+```csv
+visit_time,url,title,visit_count,typed_count
+2023-08-10 12:34:56,http://example.com,Example Site,5,2
+2023-08-09 11:22:33,http://anotherexample.com,Another Example,3,1
+```
+
+## Setting Up the Development Environment
+
+### Creating a Daytona Workspace
+
+With Daytona installed, you can create a [workspace](/definitions/d/daytona-workspace) for the [`nkkko/history` repository](https://github.com/nkkko/history):
+
+```bash
+daytona create https://github.com/nkkko/history --code
+```
+
+This command clones the [repository](/definitions/r/repository) and sets up the necessary environment. Daytona manages dependencies and configurations, ensuring that your environment is set up correctly for running the project scripts.
+
+### Set Up Environment Variables (Optional for Azure)
+
```

Comment:
We could simplify this, or even remove Azure, and maybe to a comparison of several embedding models in the process informing the reader of the best practices and differences. For example size of model vs. context legth vs. size of vector vs. time needed for embedding.

Reply:
Azure is essentially hosting OpenAI mdoels and has the same API so you can rewrite this as OpenAI API compatible hosted embedding model. The main idea is to inform the reader on the options and differences so that they could develop their own project.

Reply:
Added a comparison table between models